CesiumRasterOverlays/include/CesiumRasterOverlays/WebMapTileServiceRasterOverlay.h

```diff
@@ -0,0 +1,168 @@
+#pragma once
+
+#include "Library.h"
+#include "RasterOverlay.h"
+
+#include <CesiumAsync/IAssetRequest.h>
+#include <CesiumGeometry/QuadtreeTilingScheme.h>
+#include <CesiumGeospatial/Ellipsoid.h>
+#include <CesiumGeospatial/GlobeRectangle.h>
+#include <CesiumGeospatial/Projection.h>
+
+#include <functional>
+#include <memory>
+#include <vector>
+
+namespace CesiumRasterOverlays {
+
+class CreditSystem;
+
+/**
+ * @brief Options for tile map service accesses.
+ */
+struct WebMapTileServiceRasterOverlayOptions {
+
+  /**
+   * @brief The MIME type for images to retrieve from the server.
+   *
+   * Default value is "image/jpeg"
+   */
+  std::optional<std::string> format;
+
+  /**
+   * @brief The subdomains to use for the <code>{s}</code> placeholder in the URL template.
+   *
+   * If this parameter is a single string, each character in the string is a
+   * subdomain. If it is an array, each element in the array is a subdomain.
+   */
+  std::optional<std::variant<std::string, std::vector<std::string>>> subdomains;
```

Comment:
subdomains only accepts vector form now. if it is empty, subdomain will not be mapped into urltemplate so that only main domain will be used in requests